docs/network-flow-visibility.md

```diff
@@ -610,3 +613,75 @@ and other Theia features, please refer to the
 **Starting with Antrea v1.7, support for the ELK Flow Collector has been removed.**
 Please consider using the [Grafana Flow Collector](#grafana-flow-collector-migrated)
 instead, which is actively maintained.
+
+## L7 Visibility
+
+In addition to Layer 4 network visibility, Antrea adds Layer 7 network flow visibility
+enabling users to collect and visualize application-layer flow data in Theia.
+
+### Prerequisites
+
+To achieve L7 (Layer 7) network flow visibility, the `L7Visibility` feature gate must be enabled.
+Due to the constraint of the application detection engine, TX checksum offloading must be disabled via the
+`disableTXChecksumOffload` option in antrea-agent.conf for the feature to work. An example configuration is as below:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    disableTXChecksumOffload: true
+    featureGates:
+      L7Visibility: true
+```
+
+Alternatively, you can use the following helm installation command to configure the above options:
+
+```bash
+helm install antrea antrea/antrea --namespace kube-system --set featureGates.L7Visibility=true,disableTXChecksumOffload=true
+```
+
+### Usage
+
+To export layer 7 flows of a Pod or Namespace, user can annotate Pods or Namespaces
+with the annotation key `visibility.antrea.io/enable-l7` and set the value to indicate
+the traffic flow direction, which can be `ingress`, `egress` or `both`.
+
+For example, to enable L7 flow visibility in the ingress direction on
+Pod test-pod in the default Namespace, you can use:
+
+```bash
+kubectl annotate pod test-pod visibility.antrea.io/enable-l7=ingress
+```
+
+Based on the annotation, Flow Exporter will export the L7 flow data to
+the Flow Aggregator using the fields `l7ProtocolName` and `httpVals`.
+
+* `l7ProtocolName` field is used to store application layer protocol name,
+field will be empty if application layer flow is not exported.
+* `httpVals` stores a serialized JSON object with every HTTP flow mapped to the TxID.
+Mapping the HTTP flows to the TxID helps store all HTTP flows for a connection
+in a single row. Layer 7 flows follows the same path as other Layer 4 flows
+as elaborated above.
```

Comment:
What I meant is that for a long-lived TCP connection with multiple HTTP requests, we may end up sending multiple records from the Flow Exporter to the Flow Aggregator, and the Flow Aggregator may in turn export multiple records. In this case, are we only sending the HTTP data which is new (since the last export), or are we sending all the available HTTP data every time we export (starting at TxID 1 every time)? It feels like the first option is more efficient, and by including the TxID as the key, things are still easy to track on the receiver side.

Reply:
we will send the new HTTP data continuing with the next TxID. TxID will not overlap with the precious records.